opentracing-api/src/main/java/io/opentracing/io/opentracing/mocktracer/MockSpan.java

```diff
@@ -0,0 +1,219 @@
+package io.opentracing.io.opentracing.mocktracer;
+
+import io.opentracing.Span;
+import io.opentracing.SpanContext;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * MockSpans are created via MockTracer.buildSpan(...), but they are also returned via calls to
+ * MockTracer.finishedSpans(). They provide accessors to all Span state.
+ *
+ * @see MockTracer#finishedSpans()
+ */
+public class MockSpan implements Span {
+    // A simple-as-possible (consecutive for repeatability) id generator.
+    private static AtomicLong nextId = new AtomicLong(0);
+
+    private final MockTracer mockTracer;
+    private MockContext context;
+    private final long parentId; // 0 if there's no parent.
+    private final long startMicros;
+    private long finishMicros;
+    private final Map<String, Object> tags;
+    private final List<LogEntry> logEntries = new ArrayList<>();
+    private final String operationName;
+
+    public String operationName() {
+        return this.operationName;
+    }
+    /**
+     * TODO: Support multiple parents in this API.
+     *
+     * @return the spanId of the Span's parent context, or 0 if no such parent exists.
+     *
+     * @see MockContext#spanId()
+     */
+    public long parentId() {
+        return parentId;
+    }
+    public long startMicros() {
+        return startMicros;
+    }
+    /**
+     * @return the finish time of the Span; only valid after a call to finish().
```

Comment:
I like the assert, but I also like the javadoc for people who happen to use javadocs.